karapace/schema_reader.py

```diff
@@ -363,8 +363,14 @@ def consume_messages(self, msgs: list[Message], watch_offsets: bool) -> None:
 
                 assert message_key is not None
                 key = json_decode(message_key)
+            except AssertionError as exc:
+                LOG.warning("Empty msg.key() at offset %s", msg.offset())
+                self.offset = msg.offset()  # Invalid entry shall also move the offset so Karapace makes progress.
+                self.kafka_error_handler.handle_error(location=KafkaErrorLocation.SCHEMA_READER, error=exc)
+                continue  # [non-strict mode]
```

Comment:
Hmm, this looks like a bit of a sketchy idea, an `AssertionError` could come from deeper in the code, and this essentially abuses the `assert` to function as a `goto`. `AssertionError` should really never be caught, bar some special cases in test harnessing code.

Reply:
If an `AssertionError` is raised from `msg.key()` or `msg.error()`, the correct behavior is for the code to fall over, and be restarted by the scheduler.

Reply:
Yes this had existed prior to this change, we can take a look later.